File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/TagV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical tag row including source and hierarchy context.
 */
public record TagV1(
        String tagId,
        String name,
        String normalizedName,
        String tagType,
        String sourceSystem,
        String color,
        String description,
        long createdAtEpochMs,
        long lastUsedAtEpochMs,
        int usageCount
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/TagV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class TagV1(
    val tagId: String,
    val name: String,
    val normalizedName: String,
    val tagType: String,
    val sourceSystem: String,
    val color: String,
    val description: String,
    val createdAtEpochMs: Long,
    val lastUsedAtEpochMs: Long,
    val usageCount: Int
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
